src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -322,16 +322,26 @@ object ProtocPlugin extends AutoPlugin {
           )
         )
         .value,
-      PB.runProtoc := {
-        val s    = streams.value
-        val exec = PB.protocExecutable.value.getAbsolutePath.toString
-        (protocbridge.ProtocRunner.fromFunction { (args, extraEnv) =>
-          s.log.debug(
-            s"Executing protoc with ${args.mkString("[", ", ", "]")} and extraEnv=$extraEnv"
-          )
-          ()
-        } zip protocbridge.ProtocRunner(exec)).map(_._2)
-      },
+      PB.runProtoc := Def.taskDyn {
+        val s = streams.value
+        if (PB.targets.value == Nil) {
```

Comment:
not particularly happy of having this condition here, but extracting the noop path https://github.com/thesamet/sbt-protoc/blob/887f2c6a2f329c71ca8366de29399c6e5bfef449/src/main/scala/sbtprotoc/ProtocPlugin.scala#L454-L459 all the way to turn the source generator to a dynamic task and evaluate `PB.runProtoc` conditionally inside the `cachedCompile` was bringing a lot of changes